[KAS-8]: make Sign Up Screen

RiffatiKhan opened the pull request:
## Jira Story Link:
[KAS-8](https://leezeit.atlassian.net/browse/KAS-8)

**Description**:
1) Sign up screen with one logo, one label, four text fields and one button.
2) First and last name text fields only accept letters.
3) Next button enable only when user enter a valid email.
3) Alert shows message if any text field is empty.
4) Password text field is limited to six letters and its secure entry.
5) Key board dismiss when user tap anywhere on the screen.
6) Screen is scrollable so keyboard doesn’t  cover the text fields and button.
7) Screen is designed using a storyboard.

**How to Test this**:
Click the button in main storyboard.

**ScreenShot & Gif** 

<img width="149" alt="Screen Shot 2022-09-06 at 7 22 33 PM" src="https://user-images.githubusercontent.com/111387186/188762281-313cd4db-19f2-4ec4-84a3-ad5e8200acbb.png">

RiffatKhan committed:
Create Account Storyboard added

RiffatKhan committed:
Resolved

RiffatKhan committed:
Make Create Account Screen

haroonmaqsoodch commented on LeezeIt/LeezeIt/TextField.swift:
```diff
@@ -0,0 +1,32 @@
+extension UITextField {
+    @IBInspectable var maxLength: Int {
+        get {
+            guard let l = __maxLengths[self] else {
```
your story says "Restrict the user to enter more than 6 letters in password." change the logic of maxLength

RiffatiKhan replied:
Resolved

haroonmaqsoodch commented on LeezeIt/LeezeIt/TextField.swift:
```diff
@@ -0,0 +1,32 @@
+extension UITextField {
+    @IBInspectable var maxLength: Int {
+        get {
+            guard let l = __maxLengths[self] else {
```
what is L? fix your naming convention please

RiffatiKhan replied:
Resolved

haroonmaqsoodch commented on LeezeIt/LeezeIt/TextField.swift:
```diff
@@ -0,0 +1,32 @@
+            return l
+        }
+        set {
+            __maxLengths[self] = newValue
```
why are you use underscores before maxLength?

RiffatiKhan replied:
Resolved

haroonmaqsoodch commented on LeezeIt/LeezeIt/ViewController.swift:
```diff
@@ -9,11 +9,10 @@ import UIKit
+    @IBAction func gotoNextScreen() {
+        
+        let storyboard = UIStoryboard.init(name: "CreateAccount", bundle: nil)
+        let nextVc = storyboard.instantiateViewController(withIdentifier: "CreateAccountViewController") as! CreateAccountViewController
```
try not to force downcast

RiffatiKhan replied:
Resolved

haroonmaqsoodch commented on LeezeIt/LeezeIt/CreateAccountViewController.swift:
```diff
@@ -0,0 +1,87 @@
+        self.present(alert, animated: true,completion: nil)
+    }
+    
+    func showAlertOfMissingTextInTextField() {
```
I was not able to see alert, could attach a screen record when alert appears?

RiffatiKhan replied:
https://user-images.githubusercontent.com/111387186/188982110-39eb22d9-e551-4148-9f41-1611666285ec.mp4

RiffatKhan committed:
Issue Resolved

RiffatKhan committed:
Resolved text field issues.

haroonmaqsoodch reviewed (commented):
After typing the email and password button stays disabled untill clicked twice, check that also

RiffatKhan committed:
Resolved button enabling issue

RiffatKhan committed:
no message

RiffatKhan committed:
Resolved issues

RiffatKhan committed:
no message

RiffatiKhan committed:
Clean Up

RiffatiKhan committed:
cleanUp

RiffatiKhan committed:
cleanUp

RiffatiKhan committed:
cleanUp

RiffatiKhan committed:
Resolve Conflicts:

  [KAS-6]: clean up
  make phone number screen
  make phone number screen
  make a Phone Number screen
  Make Login Screen Programmatically.
  Make Login Screen Programmatically.

# Conflicts:
#	LeezeIt/LeezeIt.xcodeproj/project.pbxproj
#	LeezeIt/LeezeIt.xcodeproj/project.xcworkspace/xcuserdata/riffat.xcuserdatad/UserInterfaceState.xcuserstate
#	LeezeIt/LeezeIt/Base.lproj/Main.storyboard
#	LeezeIt/LeezeIt/ViewController.swift

RiffatiKhan committed:
Merge branch 'develop' into feature/KAS-8

* develop:
  Merged Develop into Kas07
  clean up
  make Four digit verification screen
  Build a phone verification screen

# Conflicts:
#	LeezeIt/LeezeIt.xcodeproj/project.pbxproj

RiffatiKhan committed:
Resolved conflicts
